src/core/component.js

```diff
@@ -172,6 +176,32 @@ Component.prototype = {
       newData: this.getData(),
       oldData: previousData
     });
+  },
+
+  /**
+   * Extends the schema of the component with a given new schema
+   * Some components might want to mutate their schema based on
+   * certain conditions. e.g: The material component changes its
+   * squema based on the selected shader to account for the
+   * different uniforms
+   * @param  {object} newSchema The schema that wil extend the original one
```

Comment:
to match other jsdoc: `@param newSchema {object} - Schema to use to extend the original one.`

can reduce words in comments by not using the word "the", and describe it as a noun.